chapter_13_database_layer_validation.asciidoc

```diff
@@ -1252,6 +1284,9 @@ That final scratchpad item will be the subject of the next chapter.
 I always like to push my validation logic down as low as possible.
 ((("model-layer validation", "benefits and drawbacks of")))
 
+// DAVID: Are you, though? This is only validation at the Python layer, not the database layer.
+// Better to test against some SQL that tries to insert it?
```

Comment:
this is why we had the discussion of `.save()` raising `IntegrityError` straight from the db, vs `.full_clean()` raising `ValidationError` without actually touching it.  let me see if i can clarify that section...

Reply:
aaaah ok i see, re-reading the chapter text now, we never actually managed to get an IntegrityError out.  We do in the next chapter!  will rephrase, thanks David.